lib/webauthn/authenticator_assertion_response.rb

```diff
@@ -33,11 +35,19 @@ def authenticator_data
     attr_reader :credential_id, :authenticator_data_bytes, :signature
 
     def valid_signature?(public_key_bytes)
-      group = OpenSSL::PKey::EC::Group.new("prime256v1")
-      key = OpenSSL::PKey::EC.new(group)
-      public_key_bn = OpenSSL::BN.new(public_key_bytes, 2)
-      public_key = OpenSSL::PKey::EC::Point.new(group, public_key_bn)
-      key.public_key = public_key
+      key =
+        if WebAuthn::AttestationStatement::FidoU2f::PublicKey.uncompressed_point?(public_key_bytes)
+          # For backwards compatibility with stored public keys with format returned by v1.10.0 or lower
```

Comment:
Thinking about a future us reading this code, it would be great if we could add more context/documentation on why did we end up supporting the signature validation of the processed version of the COSE PK, potentially used/stored by users of the gem previous to this PR release.

Reply:
Good idea